### PR TITLE
Wallet offer fees

### LINKED ISSUE #3650: Wallet should show the invitation details for the invitation to be used in an offer, in the offer inbox
## What is the Problem Being Solved?

Invitations can be very valuable assets, and so the wallet should show information about the invitation used in an offer, when the user decides to approve or disapprove the offer. (Eventually, we would like the user to be able to switch out the invitation for another one, but at the very least, disapproving the offer should be possible based on the information about the invitation.)

In the next few days, the invitation details (the amount.value for the invitation) will have a fee and an expiration as well as the usual `instance` and `installation`. We will specifically need the fee and the expiration displayed in the offer inbox, but no particular formatting is necessary. As long as the user can read the fee as a brand petname and value and the expiration as a bigint, that is fine for this stage. Even just the invitation details as a JSON string printed might be sufficient. 

"dehydrating" in the wallet which replaces objects with the user's petnames, might be useful here.

### LINKED ISSUE #3669: Wallet should create a feePurse and apply it to Zoe so that a user can pay fees
For the user to be able to pay fees, the user's wallet needs to have a feePurse with RUN in it.  

In https://github.com/Agoric/agoric-sdk/pull/3663/, Zoe will be changed to take a feePurse as the first argument to `install`, `offer`, `startInstance`, and `getPublicFacet`. #3663 has a method [`applyFeePurse`](https://github.com/Agoric/agoric-sdk/blob/7bcc956316ec3d3f75a279dda8399a3ff0cda1b3/packages/zoe/src/applyFeePurse.js) which partially applies a feePurse, producing the original Zoe API.

When Zoe is passed to the wallet, the wallet should:
1. Make a feePurse with `E(zoe).getFeePurse()`
2. Add the feePurse to the user's list of purses
3. Allow the user to transfer RUN to the feePurse (this should fall out of purse functionality in the wallet)
4. Apply the feePurse to Zoe, so that the Zoe that the wallet interacts with is the one with the feePurse applied.

### PR DESCRIPTION
**Pull request by michaelfig**
Closes #3650 and Closes #3669

UI changes are especially for @dtribble.

Wallet changes are especially for @katelynsills.

The test changes are for anybody.  (A lot needed to change because of the automatic addition of the 'RUN' issuer during wallet initialization, which bumped up the sparse int generator in the board).

### CODE REVIEW

**Comment by michaelfig** [comment]
@katelynsills I wired in the new `makeZoeKit`, but starting a solo or chain still results in:

```
Error#1: A feePurse must be provided, not undefined
Error: A feePurse must be provided, not (an undefined)
 at makeError ()
 at fail ()
 at baseAssert ()
 at assertFeePurse ()
 at ()
 at ()
```

Is there any chance that you could make a missing feePurse only an error if the charged fee amount is nonempty?  That would enable us to move forward, but there's still a lot of work to do to have the chain bootstrap somehow bypass Zoe fees for the contracts it's instantiating, etc.

**Comment by katelynsills** [comment]
> @katelynsills I wired in the new `makeZoeKit`, but starting a solo or chain still results in:
> 
> ```
> Error#1: A feePurse must be provided, not undefined
> Error: A feePurse must be provided, not (an undefined)
>  at makeError ()
>  at fail ()
>  at baseAssert ()
>  at assertFeePurse ()
>  at ()
>  at ()
> ```
> 
> Is there any chance that you could make a missing feePurse only an error if the charged fee amount is nonempty? That would enable us to move forward, but there's still a lot of work to do to have the chain bootstrap somehow bypass Zoe fees for the contracts it's instantiating, etc.

Let's not try to test end to end yet. 3322-add-fee-zoe-methods doesn't charge any fees, just asserts that a fee purse is there, so I think it's premature to put in logic to bypass charging fees. 

The WIP PR (https://github.com/Agoric/agoric-sdk/pull/3674) on top of 3322-add-fee-zoe-methods does charge fees, and takes care of the chain bootstrap process. The way it's working is that Zoe still charges fees even for the Treasury, but bootstrap process gets a payment of RUN to use for distributing to users and for launching the treasury. 

So I think there's not a lot of work to do for the bootstrap process, but the PRs aren't ready yet for end to end testing.

**Comment by michaelfig** [comment]
> So I think there's not a lot of work to do for the bootstrap process, but the PRs aren't ready yet for end to end testing.

Okay, glad to hear that.  I just got a little scared. 😨

**Comment by katelynsills** [comment]
I don't know if this would help with the testing, but if you separate the work #3650 from #3669, you could test #3650 against current master and get that approved and merged separately.

**Comment by michaelfig** [comment]
> I don't know if this would help with the testing, but if you separate the work #3650 from #3669, you could test #3650 against current master and get that approved and merged separately.

The current master already has a way to see the `invitationDetails` (a little `</>` debug button), so until I'm testing against specific Zoe fees and expirations, we don't really gain anything from this change.

**Comment by katelynsills** [comment]
> The current master already has a way to see the `invitationDetails` (a little `</>` debug button), so until I'm testing against specific Zoe fees and expirations, we don't really gain anything from this change.

The invitation details for the invitation attached an offer to be approved or not approved? Ah, if so, I did not know that.

**Comment by michaelfig** [comment]
> > The current master already has a way to see the `invitationDetails` (a little `</>` debug button), so until I'm testing against specific Zoe fees and expirations, we don't really gain anything from this change.
> 
> The invitation details for the invitation attached an offer to be approved or not approved? Ah, if so, I did not know that.

Oh, nevermind... I'm wrong.  Once this PR lands we'll have the `invitationDetails`.

File: packages/dapp-svelte-wallet/ui/src/Transaction.svelte
Comment (dtribble) [inline packages/dapp-svelte-wallet/ui/src/Transaction.svelte:138]: Why is the expiration a float?  or rather, why does this need `parseFloat`?

File: packages/vats/src/vat-zoe.js
Comment (dtribble) [inline packages/vats/src/vat-zoe.js:11]: I don't see a change corresponding to adding a 4th argument.

File: packages/dapp-svelte-wallet/ui/src/Transaction.svelte
Comment (dtribble) [reply packages/dapp-svelte-wallet/ui/src/Transaction.svelte:138]: Because it's in seconds.

File: packages/vats/src/vat-zoe.js
Comment (dtribble) [reply packages/vats/src/vat-zoe.js:11]: We discussed

**Comment by dtribble** [comment]
OK it looks good but can't be approved until the rest of Kate's changes go in.  I'll rereview then.

**Comment by katelynsills** [comment]
> OK it looks good but can't be approved until the rest of Kate's changes go in. I'll rereview then.

Oh, why is that? The base for this PR is 3294-zoe-metering, which is the base for the zoe metering changes and includes all the approved changes so far. If we approve and merge, it won't go into master, it will go into the Zoe metering changes.

**Comment by michaelfig** [comment]
> > OK it looks good but can't be approved until the rest of Kate's changes go in. I'll rereview then.
> 
> Oh, why is that? The base for this PR is 3294-zoe-metering, which is the base for the zoe metering changes and includes all the approved changes so far. If we approve and merge, it won't go into master, it will go into the Zoe metering changes.

Ah, true.  I think this just needs your approval to merge too, @katelynsills, since you have ownership of that base branch.

**Review by katelynsills — APPROVED** [review]
LGTM!

**Pull request merged**
